examples/sampler/simulated_annealing_sampler.py

```diff
@@ -0,0 +1,120 @@
+"""
+Optuna example that implements a custom relative sampler based on Simulated Annealing algorithm.
+
+Note that this implementation isn't intended to be used for production purposes and
+has the following limitations:
+- The sampler only supports `UniformDistribution` (i.e., `Trial.suggest_uniform` method).
+- The implementation prioritizes simplicity over optimization efficiency.
+
+You can run this example as follows:
+    $ python simulated_annealing_sampler.py
+
+"""
+
+import numpy as np
+
+import optuna
+from optuna import distributions
+from optuna.samplers import BaseSampler
+from optuna import structs
+
+if optuna.types.TYPE_CHECKING:
+    from typing import Any  # NOQA
+    from typing import Dict  # NOQA
+    from typing import Optional  # NOQA
+
+    from optuna.distributions import BaseDistribution  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
+    from optuna.study import InTrialStudy  # NOQA
+
+COOLDOWN_FACTOR = 0.9
+NEIGHBOR_RANGE_FACTOR = 0.1
+
+
+class SimulatedAnnealingSampler(BaseSampler):
+    def __init__(self, temperature=100, seed=None):
+        # type: (int, Optional[int]) -> None
+
+        self._rng = np.random.RandomState(seed)
+        self._independent_sampler = optuna.samplers.RandomSampler(seed=seed)
+        self._temperature = temperature
+        self._current_params = {}  # type: Dict[str, Any]
+
+    def infer_relative_search_space(self, study, trial):
+        # type: (InTrialStudy, FrozenTrial) -> Dict[str, BaseDistribution]
+
+        return optuna.samplers.product_search_space(study)
+
+    def sample_relative(self, study, trial, search_space):
+        # type: (InTrialStudy, FrozenTrial, Dict[str, BaseDistribution]) -> Dict[str, Any]
+
+        if search_space == {}:
+            return {}
+
+        prev_trial = self._last_complete_trial(study)
+        if self._rng.uniform(0, 1) <= self._probability(study, prev_trial.value, study.best_value):
```

Comment:
In my understanding, the transition probability is calculated by using the current value instead of the best value. I think there are some variations, and could you show us the reference for this implementation?

Reply:
I'm sorry. The current implementation is wrong. It should use the current value instead of the best value.

Reply:
Fixed: https://github.com/pfnet/optuna/pull/446/commits/bd123c14811d906406a3d0a5ef8214356864b579